contracts/callpaths/WarmPath.sol

```diff
@@ -228,4 +329,41 @@ contract WarmPath is MarketSequencer, SettleLayer, ProtocolAccount {
                code == UserCmd.BURN_AMBIENT_QUOTE_LP ||
                code == UserCmd.HARVEST_LP;
     }
+
+    /// @notice Validate that the pool is stable swap and bid and ask ticks are in correct range for providing concentrated liquidity.
+    /// @dev This is a helper function to validate the concentrated liquidity minting and burning.
+    /// @param base The base token address.
+    /// @param quote The quote token address.
+    /// @param poolIdx The index of the pool type.
+    /// @param bidTick The price tick index of the lower boundary of the range order.
+    /// @param askTick The price tick index of the upper boundary of the range order.
+    /// @param pool The pool specification.
+    function _validateConcentratedLiq (address base, address quote, uint256 poolIdx, int24 bidTick, int24 askTick, PoolSpecs.Pool memory pool) internal view {
+        if (poolIdx == stableSwapPoolIdx_) {
+
+            // get the decimal of base and quote token.
+            uint8 baseTokenDecimal = ERC20(base).decimals();
+            uint8 quoteTokenDecimal = ERC20(quote).decimals();
```

Comment:
I don't know if we should assume that all tokens incoming implement `decimals()`.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/IERC20.sol

The IERC20Minimal.sol in interfaces also doesn't assume decimals() exist.

I do think we should try to query for decimals, and if it doesn't exist -- default to 18